infinispan-client/src/test/java/io/quarkus/ts/openshift/infinispan/client/InfinispanCountersOpenShiftIT.java

```diff
@@ -1,328 +1,238 @@
+                .statusCode(Matchers.allOf(Matchers.greaterThanOrEqualTo(500), Matchers.lessThan(600)));
 
         restartInfinispanCluster();
 
         // try to connect back to infinispan cluster and expect no content
         await().atMost(5, TimeUnit.MINUTES).untilAsserted(() -> {
             when()
-                    .get("/first-counter/get-cache")
+                    .get(appUrl + "/first-counter/get-cache")
                     .then()
                     .statusCode(204);
         });
 
         // check the client counter
-        when()
-                .get("/first-counter/get-client")
-                .then()
-                .body(is("11"));
+        String clientCounter = getCounterValue(appUrl + "/first-counter/get-client");
+        assertEquals("11", clientCounter);
     }
 
-    private void resetCounters() {
-        when()
-                .put("/first-counter/reset-cache")
-                .then()
-                .body(is("Cache=0"));
+    @Test
+    @Order(8)
+    public void testConnectSecondClient() throws OpenShiftTestException {
+        resetCacheCounter(appUrl + "/first-counter/reset-cache");
 
-        when()
-                .put("/first-counter/reset-client")
-                .then()
-                .body(is("Client=0"));
+        String secondClientCache = getCounterValue(openshift.getUrlFromRoute(SECOND_CLIENT_APPLICATION_NAME) + "first-counter/get-cache");
+        assertEquals("0", secondClientCache);
     }
 
-    private void incrementCountersOnValue(int count) {
-        for (int i = 1; i <= count; i++) {
-            when()
-                    .put("/first-counter/increment-counters")
-                    .then()
-                    .body(is("Cache=" + i + " Client=" + i));
-        }
+    @Test
+    @Order(9)
+    public void testMultipleClientIncrement() throws OpenShiftTestException {
+        resetCacheCounter(appUrl + "/first-counter/reset-cache");
+        resetClientCounter(appUrl + "/first-counter/reset-client");
+
+        // fill the cache in first and second client
+        incrementCountersOnValue(appUrl + "/first-counter/increment-counters", 10);
+        incrementCountersOnValue(openshift.getUrlFromRoute(SECOND_CLIENT_APPLICATION_NAME) + "first-counter/increment-counters", 10);
+
+        // save the cache counters in first and second client
+        String firstClientCacheCounter = getCounterValue(appUrl + "/first-counter/get-cache");
+        String secondClientCacheCounter = getCounterValue(openshift.getUrlFromRoute(SECOND_CLIENT_APPLICATION_NAME) + "first-counter/get-cache");
+
+        // save the client counters in first and second client
+        String firstClientAppCounter = getCounterValue(appUrl + "/first-counter/get-client");
+        String secondClientAppCounter = getCounterValue(openshift.getUrlFromRoute(SECOND_CLIENT_APPLICATION_NAME) + "first-counter/get-client");
+
+        assertEquals("20", firstClientCacheCounter);
```

Comment:
The PR is WIP because sometimes (about 50 % chance) I got the value "21" in here instead of "20". The "for loop", that fills the cache and client counters, works well in other tests before but here it sometimes somehow adds one more cache to it. Just to the cache counter, the client counter remains the same. I'm investigating it.